### ISSUE TITLE
Modules which cannot be accessed are not included in projects.json

### ISSUE THREAD

**Issue by CIAvash** [perl6/ecosystem#397]
If a module cannot be reached because the host server is down, the module won't be included in [projects1.json](http://ecosystem-api.p6c.org/projects1.json) and [projects.json](http://ecosystem-api.p6c.org/projects.json). And the module manager(zef) would not be able to find the module, although the module exists, it just could not be accessed.

Apparently projects1.json is generated in [server/updatelist.pl](https://github.com/perl6/ecosystem/blob/master/server/updatelist.pl) and projects.json is generated in [updatelist.pl](https://github.com/perl6/ecosystem/blob/master/updatelist.pl).

It seems zef has 4 [mirrors](https://github.com/ugexe/zef/blob/5e1bd4c17aa733a124d0c9203aba7729c759e83a/resources/config.json#L52-L57) for getting the list of modules, first two of which are projects1.json and projects.json.

[Related IRC logs](http://colabti.org/irclogger/irclogger_log/perl6?date=2018-06-12#l749)

**Comment by moritz** [comment]
What's the point of offering a module for installation when it cannot be installed?

I consider this a feature.

**Comment by CIAvash** [comment]
Because it misleads the user. If they try to install a module and zef tells them it cannot find it, then they think it might have been removed or something. They might not try to install it again. But if zef shows an error that tells the user it cannot access it(for example "500 internal server error"), then they know there is something wrong with the server.

Yesterday I tried to install my own module and I was really confused, I thought maybe somebody accidentally removed it from the list. Then After some investigation I found out that Gitlab was down and that was what caused it.

**Comment by rafaelschipiura** [comment]
You mean that the error messages should be more specific?

**Comment by CIAvash** [comment]
Yes, but It's not just the error message. If projects1.json(the generated list) is incomplete, the only thing zef can say is that it cannot find the module.

**Comment by JJ** [comment]
Not clear what could be the solution here. Any suggestion?

**Comment by CIAvash** [comment]
One solution was offered by Zoffix: "probably could add a conditional in [here](https://github.com/perl6/ecosystem/blob/master/updatelist.pl#L14-L19) that would read use dist's info from previous run if we fail to fetch the new one".

But there are 2 scripts generating 2 different lists. And then there are two other mirrors zef uses, I don't know if they have this problem or not.

cc @ugexe
